code/src/qik_2s12v10.cpp

```diff
@@ -37,36 +115,95 @@ namespace r2d2::moving_platform {
         hwlib::wait_ms(10);
 
         // setup the uart communication
-        usart_bus << qik_autodetect_baudrate;
+        usart_bus << qik_2s12v10_registers::autodetect_baudrate;
+    }
+
+    void qik_2s12v10_c::brake(const int8_t brake_amount) {
+        usart_bus << qik_2s12v10_registers::set_motor_m0_brake << brake_amount;
+        usart_bus << qik_2s12v10_registers::set_motor_m1_brake << brake_amount;
     }
 
-    void qik_2s12v10_c::brake() {
-        set_m0_speed(0);
-        set_m1_speed(0);
+    void qik_2s12v10_c::brake_m0(const int8_t brake_amount) {
+        usart_bus << qik_2s12v10_registers::set_motor_m0_brake << brake_amount;
+        uint8_t speed = 0;
+        usart_bus << qik_2s12v10_registers::set_motor_m0_forward << speed;
     }
 
-    uint8_t qik_2s12v10_c::get_error() {
+    void qik_2s12v10_c::brake_m1(const int8_t brake_amount) {
+        usart_bus << qik_2s12v10_registers::set_motor_m1_brake << brake_amount;
+        uint8_t speed = 0;
+        usart_bus << qik_2s12v10_registers::set_motor_m1_forward << speed;
```

Comment:
We have a function to set the motor speed? Why is this not being used?

Reply:
fixed in https://github.com/R2D2-2019/moving_platform/commit/337120c219ad634bd5b2ff38cca7a77344aa2c2d